test/tests/functional/pbs_fairshare.py

```diff
@@ -351,3 +351,29 @@ def test_pbsfs(self):
         job_order = [jid3, jid4]
         for i in range(len(job_order)):
             self.assertEqual(job_order[i].split('.')[0], c.political_order[i])
+
+    def test_fairshare_decay_min_usage(self):
+        """
+        Test that fairshare decay doesn't reduce the usage below 1
+        """
+        self.scheduler.set_sched_config({'fair_share': 'True'})
+        self.scheduler.set_sched_config({'log_filter': 0})
+        self.scheduler.add_to_resource_group(TEST_USER, 10, 'root', 50)
+        self.scheduler.set_fairshare_usage(TEST_USER, 1)
+        self.scheduler.set_sched_config({"fairshare_decay_time": "00:00:02"})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
```

Comment:
Why do you set scheduling to False here and below? When there is no workload then just setting scheduling to True should do it, right?

Reply:
I set it to False because PTL makes sure that the sched cycle is over before returning control back. So, this is just to make sure that the sched cycle is over before we proceed. I actually think we should enhance PTL to just do this for scheduling=True, but that'd be out of scope for this PR.

Reply:
I don't think that is necessary. Your test setup has nothing in it (no jobs & reservation). Your test sleeps for 2 seconds between the cycles, and then there is a log match which will be attempted 5 times. I think there is enough time to complete empty cycles.

Reply:
hmm .. ok, I guess I was being a bit pedantic, I'll remove the turning scheduling off. But I think I'll remove the 5 attempts and let it be default, that way it's still guaranteed to run ok on slower machines.